packages/dev/core/src/Gizmos/gizmo.ts

```diff
@@ -368,6 +375,22 @@ export class Gizmo implements IGizmo {
         }
     }
 
+    /**
+     * if transform has a pivot and is not using PostMultiplyPivotMatrix, then the worldMatrix contains the pivot matrix (it's not cancelled at the end)
+     * so, when extracting the world matrix component, the translation (and other components) is containing the pivot translation.
+     * And the pivot is applied each frame. Removing it anyway here makes it applied only in computeWorldMatrix.
+     * @param transform local transform that needs to be transform by the pivot inverse matrix
+     * @param localMatrix local matrix that needs to be transform by the pivot inverse matrix
+     * @returns matrix transformed by pivot inverse if the transform node is using pivot without using post Multiply Pivot Matrix
+     */
+    protected _handlePivotMatrixInverse(transform: TransformNode, localMatrix: Matrix): Matrix {
+        if (transform.isUsingPivotMatrix() && !transform.isUsingPostMultiplyPivotMatrix()) {
+            transform.getPivotMatrix().invertToRef(TmpVectors.Matrix[5]);
+            TmpVectors.Matrix[5].multiplyToRef(localMatrix /*this._attachedNode._worldMatrix*/, TmpVectors.Matrix[4]);
+            return TmpVectors.Matrix[4];

```

Comment:
It's a bit dangerous to return a `TmpVectors` element from a method...

I think it would be safer to pass a ref matrix to `_handlePivotMatrixInverse` and to store the result in this matrix. This way, it's the caller responsibility to use a `TmpVectors` element or not.

Reply:
Fixed